src/main/java/com/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,108 @@
+        String query = "INSERT INTO products (name, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query,
+                    Statement.RETURN_GENERATED_KEYS);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                item.setId(resultSet.getLong(1));
+            }
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product not created", e);
+        }
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        Product product = null;
+        String query = "SELECT * FROM products WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                product = getProductFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Product with ID" + id + " not found", e);
+        }
+        return Optional.ofNullable(product);
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products;";
+        List<Product> products = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                Product product = getProductFromResultSet(resultSet);
+                products.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Could not get the list of products", e);
+        }
+        return products;
+    }
+
+    @Override
+    public Product update(Product item) {
+        String query = "UPDATE products SET name = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(query);
+            statement.setString(1, item.getName());
+            statement.setDouble(2, item.getPrice());
+            statement.setLong(3, item.getId());
+            statement.executeUpdate();
+            return item;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update product from ID "
+                    + item.getId(), e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE products SET deleted = 1 WHERE id = ?";
```

Comment:
```suggestion
        String query = "UPDATE products SET deleted = TRUE WHERE id = ?";
```
this way it would easier to understand what's going one (that goes to all statements, where deleted is used)